src/Entities/Player.cpp

```diff
@@ -1604,12 +1604,18 @@ unsigned int cPlayer::AwardAchievement(const eStatistic a_Ach)
 
 void cPlayer::TeleportToCoords(double a_PosX, double a_PosY, double a_PosZ)
 {
+	// Clamp the positions to exactly representable single-precision floating point values
+	// This is necessary to avoid rounding errors in the noise generator and overflows in the chunk loader
+	const double ClampedPosX = Clamp(a_PosX, -167770000.0, 167770000.0);
+	const double ClampedPosY = Clamp(a_PosY, -167770000.0, 167770000.0);
```

Comment:
This doesn't make that much sense.
You could for example teleport  to y = `-167770000` and then the falling mechanics should get us into bad numbers.
Clamp it to 0 - `MAX_HEIGHT` makes a lot more sense.